cmd/main.go

```diff
@@ -98,7 +102,30 @@ func main() {
 		c.NextProtos = []string{"http/1.1"}
 	}
 
-	tlsOpts := []func(*tls.Config){}
+	// Metrics endpoint is enabled in 'config/default/kustomization.yaml'. The Metrics options configure the server.
+	// More info:
+	// - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.18.4/pkg/metrics/server
+	// - https://book.kubebuilder.io/reference/metrics.html
+	metricsServerOptions := metricsserver.Options{
+		BindAddress:   metricsAddr,
+		SecureServing: secureMetrics,
+		// TODO(user): TLSOpts is used to allow configuring the TLS config used for the server. If certificates are
+		// not provided, self-signed certificates will be generated by default. This option is not recommended for
+		// production environments as self-signed certificates do not offer the same level of trust and security
+		// as certificates issued by a trusted Certificate Authority (CA). The primary risk is potentially allowing
+		// unauthorized access to sensitive metrics data. Consider replacing with CertDir, CertName, and KeyName
+		// to provide certificates, ensuring the server communicates using trusted and secure certificates.
+		TLSOpts: tlsOpts,
+	}
```

Comment:
Is it okay for us to use self-signed certs for now? Are there potential risks if the vpa-operator's metrics were ever exposed?

Reply:
I don't think that there is anything particularly sensitive in the metrics. The token itself (used to authenticate against the metrics endpoint) is potentially sensitive as it might be more privileged than just a token from a single-use service account, so it's probably a good idea to do this right.

At the same time, doing it right on a non-OpenShift cluster is rather difficult. What do you think of this idea:
- We annotate the service object such that OpenShift will create a service cert for us
- We mount the CM & secrets that OpenShift should create, but we make the mount optional so that the pod can start even if running on a non-OpenShift cluster
- We check to see if the cert & key exist at runtime in the operator, and if so, we pass them in, and if not, we set the options to generate a temporary self-signed cert

That seems like a reasonable compromise:
- On OpenShift, it just works securely out of the box
- On non-OpenShift (if anybody is using it there), it works out of the box, but less securely
- On non-OpenShift, someone can make it fully secure by creating a legit CA-signed cert and placing them where OpenShift would have placed them, and we'd start using their cert.

Reply:
Sure that sounds reasonable to me, and your explanation makes sense. service-ca operator makes this simpler. Sorry for opening pandora's box with suggesting this upgrade this late :stuck_out_tongue: